README.md

```diff
@@ -12,31 +12,51 @@ crazy stuff we might as well go and use the far more sensible practice from
 Rails 4 of storing the flash as basic ruby types and sweeping the flash before
 persisting into the session.
 
-For more details of the how and why, check out our blog post 
-[Happily upgrading Ruby on Rails at production scale](http://webuild.envato.com/blog/upgrading-ruby-on-rails-at-production-scale/)
+For more details of the how and why, check out our blog post [Happily upgrading
+Ruby on Rails at production scale][blog-post]
 
-When using this gem on a Rails 2 or 3 app:
+    [blog-post]: http://webuild.envato.com/blog/upgrading-ruby-on-rails-at-production-scale/
 
- - Flash messages are stored as basic objects in the Rails 4 style, pre-swept.
- - Flash messages in the Rails 2 format can be successfully decoded.
- - Flash messages in the Rails 3 format can be successfully decoded.
- - Flash messages in the Rails 4 format can be successfully decoded.
+When using this gem on a Rails 2.3, 3.1+ or 4 app:
 
-Additionally, on Rails 2 we include some patches for the SessionHash and
-CookieStore in order to make them act more like a HashWithIndifferentAccess,
-like the versions on Rails 3, so that your session_id can survive a trip to
-Rails 3 and back.
+ * Flash messages are stored as basic objects in the Rails 4 style, pre-swept.
+ * Flash messages in the Rails 2 format can be successfully decoded.
+ * Flash messages in the Rails 3.1 format can be successfully decoded.
+ * Flash messages in the Rails 4 format can be successfully decoded.
 
 This actually makes it possible to bounce requests from a Rails 2 server, to a
 Rails 3 server and back again so long as both servers are using this gem. Very
 helpful when you're doing a big Rails 2 => 3 upgrade and want to run a few
 Rails 3 servers concurrently with your Rails 2 cluster to verify everything is
-fine and performance is acceptable without
-having to do the all-in switch.
+fine and performance is acceptable without having to do the all-in switch.
+
+### Rails 2
+
+Additionally, on Rails 2 we include some patches for the SessionHash and
+CookieStore in order to make them act more like a HashWithIndifferentAccess,
+like the versions on Rails 3, so that your session_id can survive a trip to
+Rails 3 and back.
+
+### Rails 3.0
+
+Rails 3.0 was a weird half-way house between 2.x and 3.1. The 3.0 to 3.1
+upgrade gave many of us an [Argument Error (dump format error)
+problem][argumenterror-so].
+
+    [argumenterror-so]: http://stackoverflow.com/questions/9120501/what-causes-the-argumenterror-dump-format-error
+
+On Rails 3.0, rails 3.0 flashes can be successfully decoded, but the other
```

Comment:
No this means that when running in rails 3.0 then a flash encoded by vanilla rails 3.0 can be decoded, but if that wasn't clear then I should reword this sentence.